### PR TITLE
fix for HTTP Trigger routes overriding /admin/ APIs if route set to {*route}

### LINKED ISSUE #6990: HTTP Trigger routes can overwrite /admin/ APIs
#### Repro steps
1. Deploy an app with the below content.
2. Make an HTTP trigger request with the master key against /admin/host/status and notice how it returns the response from the HTTP trigger.

host.json
```
{
  "version": "2.0",
  "logging": {
    "applicationInsights": {
      "samplingExcludedTypes": "Request",
      "samplingSettings": {
        "isEnabled": true
      }
    }
  },
  "extensions": {
    "http": {
      "routePrefix": ""
    }
  }
```

HTTP Trigger
```csharp
        [FunctionName("Function1")]
        public static async Task<IActionResult> Run(
            [HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = "{*route}")] HttpRequest req,
            ILogger log)
        {
            log.LogInformation("C# HTTP trigger function processed a request.");

            string name = req.Query["name"];

            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
            dynamic data = JsonConvert.DeserializeObject(requestBody);
            name = name ?? data?.name;

            string responseMessage = string.IsNullOrEmpty(name)
                ? "This HTTP triggered function executed successfully. Pass a name in the query string or in the request body for a personalized response."
                : $"Hello, {name}. This HTTP triggered function executed successfully.";

            return new OkObjectResult(responseMessage);
        }
```

#### Expected behavior

Http triggers should take a lower priority in routing logic vs APIs defined by the Functions host, such as admin and extension webhook urls.

#### Actual behavior

All HTTP requests get routed to the HTTP trigger

#### Known workarounds

Pin to a previous version of the Functions Host runtime by setting the app setting `FUNCTIONS_EXTENSION_VERSION` to 3.0.14916

### PR DESCRIPTION
**Pull request by safihamid**
resolves https://github.com/Azure/azure-functions-host/issues/6990
### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

* [X] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [X] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [X] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [X] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Routing/ScriptRouteHandler.cs
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Routing/ScriptRouteHandler.cs]: So the regression was in https://github.com/Azure/azure-webjobs-sdk-extensions/pull/668 where this issue was actually discussed? The /admin guard was removed in that PR.

Comment (mathewc) [reply src/WebJobs.Script.WebHost/Routing/ScriptRouteHandler.cs]: And previously we had added that code to address this very issue (fix for https://github.com/Azure/azure-functions-host/issues/2590)

File: test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/WarmupFunctionEndToEndTests.cs
Comment (mathewc) [inline test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/WarmupFunctionEndToEndTests.cs]: Can you add a positive check here like you did in the test above? E.g. look for myversion, or preferably verify that the body contains status details #Resolved

File: src/WebJobs.Script.WebHost/Routing/ScriptRouteHandler.cs
Comment (safihamid) [reply src/WebJobs.Script.WebHost/Routing/ScriptRouteHandler.cs]: yes I actually needed to add that back and exclude admin/warmup. see other PR I just sent: https://github.com/Azure/azure-webjobs-sdk-extensions/pull/696

if we don't do it there then the status code will be 200 instead of 401 even though no http trigger function gets executed.

File: src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj
Comment (ConnorMcMahon) [inline src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj]: This appears to be the only source code behavior change now. What exactly is in this version of the HTTP extension that fixes this behavior?

Comment (safihamid) [reply src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj]: Take a look at the other PR I sent for that extension

File: src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs
Comment (ConnorMcMahon) [inline src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs:87]: This fix feels better to me! That should cover any APIs we define, as opposed to just /admin/ APIs.

**Review by ConnorMcMahon — APPROVED** [review]
LGTM!

**Comment by fabiocav** [comment]
@safihamid are you waiting for additional reviews on this?

**Comment by safihamid** [comment]
@fabiocav I was hoping to get one more approval on this from either you or @mathewc in case you can think of something that this might impact.

**Review by mathewc — APPROVED** [review]

**Pull request merged**
